### ISSUE TITLE
Can't read profile data on Android device: Error: ENOENT: no such file or directory, open 'chrome-profile-results-...

### ISSUE THREAD

**Issue by joakin** [axemclion/browser-perf#62]
Hi!

I'm trying to benchmark a wikipedia page on my android device. When running

```
browser-perf https://en.m.wikipedia.org/wiki/Barack_Obama --selenium=localhost:9515 --config-file=conf.json
```

With conf.js being:

```json
{
  "selenium": "http://localhost:9515",
  "browsers": [{
    "browserName": "android"
  }]
}
```

...having chromedriver running, I get a new browser window that loads the page, I see a toast that notifies of the start of the profile recording, the page scrolls to the bottom, and I see a notification about the profile finished recording.

Then on my local machine I get the following exception:

```sh
browser-perf https://en.m.wikipedia.org/wiki/Barack_Obama --selenium=localhost:9515 --config-file=conf.json
[ { Error: ENOENT: no such file or directory, open 'chrome-profile-results-2016-05-20-090003'
      at Error (native)
      at Object.fs.openSync (fs.js:634:18)
      at Object.fs.readFileSync (fs.js:502:33)
      at /usr/local/lib/node_modules/browser-perf/lib/probes/AndroidTracingProbe.js:169:29
      at _fulfilled (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:834:54)
      at self.promiseDispatch.done (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:863:30)
      at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:796:13)
      at /usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:604:44
      at runSingle (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:137:13)
      at flush (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:125:13)
    errno: -2,
    code: 'ENOENT',
    syscall: 'open',
    path: 'chrome-profile-results-2016-05-20-090003' } ]
Error: ENOENT: no such file or directory, open 'chrome-profile-results-2016-05-20-090003'
    at Error (native)
    at Object.fs.openSync (fs.js:634:18)
    at Object.fs.readFileSync (fs.js:502:33)
    at /usr/local/lib/node_modules/browser-perf/lib/probes/AndroidTracingProbe.js:169:29
    at _fulfilled (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:834:54)
    at self.promiseDispatch.done (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:863:30)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:796:13)
    at /usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:604:44
    at runSingle (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:137:13)
    at flush (/usr/local/lib/node_modules/browser-perf/node_modules/q/q.js:125:13)
```

It doesn't seem like anything went wrong on the device, and from what I [see on the code](https://github.com/axemclion/browser-perf/blob/master/lib/probes/AndroidTracingProbe.js#L167) it seems like a problem with adb accessing the device storage.

Any ideas?

Some more detailed info:
* Nexus 5
* ChromeDriver 2.21.371459
* Chrome Mobile 50.0.2661.89
* Android 6.0.1; Nexus 5 Build/MOB30H

**Comment by joakin** [comment]
Here's the log with `--verbose` https://gist.github.com/joakin/1bd3b2aa2a86e73cedb4df86bc855cfe

**Comment by axemclion** [comment]
@joakin I ran this with 
`browser-perf https://en.m.wikipedia.org/wiki/Barack_Obama --selenium=localhost:9515 --browsers=android` and I was able to see all the data. I am on

- Android 5.1.1
- Chrome 48.0.2564.95

Could this be due to a version mismatch ? Can you see if you have a file called `/sdcard/Download/chrome-profile-results-2016-05-20-093039` that was created on the device ?

**Comment by axemclion** [comment]
@joakin Can you also add adb logs from the test runs ? That will tell us if there is a file that gets created.

**Comment by joakin** [comment]
I've just tried, there is no `/sdcard/Download/chrome-profile-results*` on the device after a run with the same error

```
$ adb shell
shell@hammerhead:/ $ find /sdcard/ -type d -name 'Down*' 2>/dev/null
/sdcard/Download
/sdcard/Android/data/org.mozilla.firefox/files/Download
shell@hammerhead:/ $ find / -type d -name 'chrome-profile-results*' 2>/dev/null
shell@hammerhead:/ $ find /sdcard/ -type d -name 'chrome-profile-results*' 2>/dev/null
shell@hammerhead:/ $ find /sdcard/ -type d -name 'chrome-profile*' 2>/dev/null
shell@hammerhead:/ $
```

No luck.

Manually `adb pull`ing files from the device works:

```
browser-perf-test $ adb shell
shell@hammerhead:/ $ mkdir /sdcard/Download/test/
shell@hammerhead:/ $ touch $_/test.txt
shell@hammerhead:/ $ ls /sdcard/Download/test/
test.txt
shell@hammerhead:/ $ ^D
browser-perf-test $ adb pull -p /sdcard/Download/test
pull: building file list...
pull: /sdcard/Download/test/test.txt -> ./test.txt
1 file pulled. 0 files skipped.
browser-perf-test $ ls
conf.json run.sh    test.txt
```

I'm going to find out how to get the adb logs and I'll post back.

**Comment by joakin** [comment]
I've run `adb logcat` and run the test again, the relevant portion of the log I'm pasting below, it seems like it can't open `/storage/emulated/0/Download/...`

Which is weird since I've downloaded files recently from my Chrome sessions. I'm going to try and download stuff manually from the browser after a run (with the clean profile and logged out).

```
05-24 10:21:08.008 16995 16995 I chromium: [INFO:CONSOLE(129)] "\o/ Hey! This is open source software and we need volunteers to help us build this thing, make it better and fix any bugs that you might be seeing in this JavaScript console!
05-24 10:21:08.008 16995 16995 I chromium:
05-24 10:21:08.008 16995 16995 I chromium: You can find our backlog @ https://phabricator.wikimedia.org/project/profile/67/", source: https://en.m.wikipedia.org/w/load.php?debug=false&lang=en&modules=ext.gadget.switcher%7Cext.navigationTiming%2CwikimediaEvents%7Cext.visualEditor.targetLoader%2Ctrack%2Cve%7Cjquery.throttle-debounce%7Cmediawiki.toc%7Cmediawiki.ui.input%7Cmobile.ajax%2Cbetaoptin%2Cdrawers%2Cissues%2CmodifiedBar%2Coverlays%2Cpagelist%2Creferences%2Csearch%2Csettings%2Csite%2Cstartup%2Ctoast%2Ctoggle%2Cwatchstar%7Cmobile.editor.api%7Cmobile.issues.images%7Cmobile.loggingSchemas.edit%2CmobileWebLanguageSwitcher%2CmobileWebSearch%7Cmobile.overlay.images%7Cmobile.pagelist.scripts%2Cstyles%7Cmobile.pagesummary.styles%7Cmobile.references.gateway%2Cimages%7Cmobile.toggle.images%7Cschema.Edit%2CMobileWebLanguageSwitcher%2CMobileWebSearch%2CNavigationTiming%2CSaveTiming%2CUniversalLanguageSelector%7Cskins.minerva.editor%2Cscripts%2Ctoggling%2Cwatchstar%7Cskins.minerva.icons.images.scripts&skin=minerva&version=8a8a8d511116 (129)
05-24 10:21:09.435 16995 16995 I chromium: [INFO:CONSOLE(281)] "Scrolling with ", source:  (281)
05-24 10:21:16.409 17297 17297 D AndroidRuntime: >>>>>> START com.android.internal.os.RuntimeInit uid 2000 <<<<<<
05-24 10:21:16.413 17297 17297 D AndroidRuntime: CheckJNI is OFF
05-24 10:21:16.454 17297 17297 D ICU     : No timezone override file found: /data/misc/zoneinfo/current/icu/icu_tzdata.dat
05-24 10:21:16.494 17297 17297 I Radio-JNI: register_android_hardware_Radio DONE
05-24 10:21:16.513 17297 17297 D AndroidRuntime: Calling main entry com.android.commands.am.Am
05-24 10:21:16.536 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.537 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.542 17297 17297 D AndroidRuntime: Shutting down VM
05-24 10:21:16.544 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.547 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.558 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.564 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.569 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.581 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.581 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.587 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.592 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.594 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.600 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.603 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.614 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.620 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.623 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.628 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.633 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.638 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.639 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.644 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.645 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.649 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.655 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.660 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.666 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.671 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.676 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.682 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.687 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.692 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.702 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.703 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.709 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.714 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.719 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.725 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.730 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.736 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.741 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.747 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.752 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.757 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.763 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.768 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.773 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.779 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.784 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.790 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.795 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.801 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.807 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.812 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.818 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.823 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.828 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.834 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.840 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.845 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.850 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.856 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.861 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.866 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.889 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.890 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.891 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.892 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.894 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.899 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.905 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.911 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.916 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.916 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.927 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.937 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.948 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.957 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.967 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.978 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.987 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.992 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:16.997 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.002 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.007 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.012 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.017 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.022 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.027 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.032 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.037 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.042 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.047 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.052 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.056 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.062 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.066 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.071 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.076 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.081 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.086 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.091 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.096 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.100 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.104 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.110 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.110 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.110 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.111 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.111 16995 17063 E chromium: [ERROR:tracing_controller_impl_data_sinks.cc(92)] Failed to open /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107
05-24 10:21:17.112 16995 16995 I cr_TracingController: Profiler finished. Results are in /storage/emulated/0/Download/chrome-profile-results-2016-05-24-082107.
```

**Comment by joakin** [comment]
Okay, so here's something weird. After I run the test and it fails, I go to the Chrome app that was open by the test, and I try to download barack's image, and it prompts me with the permissions dialog for storage access.

![screen shot 2016-05-24 at 10 40 25](https://cloud.githubusercontent.com/assets/657839/15497717/533432d2-219c-11e6-9336-9273df71c372.png)
![screen shot 2016-05-24 at 10 40 46](https://cloud.githubusercontent.com/assets/657839/15497716/533333d2-219c-11e6-87f0-176058052431.png)

After accepting those, it saves the file just fine.

If I immediately re-launch the test, it seems to open a new chrome without the storage access permission, so it fails again (even if it says it went alright)

![screen shot 2016-05-24 at 10 41 49](https://cloud.githubusercontent.com/assets/657839/15497718/533438cc-219c-11e6-8116-f5439ff7d1cc.png)

I'm not sure what to do next.

**Comment by axemclion** [comment]
This could either be an issue with Android 6.0 permissions, or with the way your storage is configured. Let me see if I can get my hands on an Android 6.0 device and try running this. 
In the meantime, what are the permissions on your sdcard storage device ?

**Comment by axemclion** [comment]
I finally got my hands on an Android 6.0 device, and looks like this is an issue with the permissions for the app. If you go to Settings > Apps > Chrome and look at the permissions, Chrome does not have __Storage__ permissions. This is the reason the log trace file is not written. 
I am now investigating ways to make Chrome have storage permissions by default.

**Comment by axemclion** [comment]
So, looks like we need to run 

```
adb shell pm grant com.android.chrome android.permission.WRITE_EXTERNAL_STORAGE
adb shell pm grant com.android.chrome android.permission.READ_EXTERNAL_STORAGE
```

before starting to collect the trace files. I will look at adding this to the code. I am still noticing that the framerates are around 120 - looks like something is being double counted. Will look into that next.

**Comment by axemclion** [comment]
@joakin I fixed this bug, and release browser-perf@1.4.7. 

I also tried it on my android device and it works good. Closing the issue as fixed. Please reopen if you still have trouble.

**Comment by joakin** [comment]
@axemclion Awesome! I just tried it and it worked great.

I'm not seeing weird fps numbers, below 60 for me on that example ([output](https://gist.github.com/joakin/c900fae8eddc57d95ceedeb1892aac80))

Thanks a lot!
